### DISCUSSION TITLE
Request: Ability to use non-image content in carousel

### DISCUSSION THREAD

**Discussion by atyshka** [bootstrap-vue/bootstrap-vue#6051] [Q&A] [answered]
### Is your feature request related to a problem? \

Currently there is no supported way to make a carousel with non-image content. I would like to make a carousel that contains b-card components but cannot find a way to do this.

### Describe the solution you'd like

First class support for custom content via a slot, or a workaround that can use the image slot current available.

### Describe alternatives you've considered

Tried using the img slot for the carousel slide, but cannot get sizing to work correctly. I may resort to vanilla bootstrap carousel which seems more generic.

**Comment by Hiws** [comment, marked as answer]
The `img` slot, can take any content and not just images. It's just poorly named.

```html
<b-carousel>
  <b-carousel-slide>
    <template #img>
      <b-card>
        <b-card-title>Title {{ i }}</b-card-title>
        <b-card-sub-title>Sub-title</b-card-sub-title>
        <b-card-text>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin congue massa felis, finibus dignissim augue lacinia id. Proin efficitur leo massa, eget faucibus est rhoncus eget.
        </b-card-text>
      </b-card>
    </template>
  </b-carousel-slide>
</b-carousel>
```

Live example:
<https://codepen.io/Hiws/pen/eYzXreV>

**Comment by atyshka** [comment]
Thanks for sharing! Your codepen looks good, but when I embed in my site I have a weird glitch where when the animation is in progress, all of my bootstrap columns, even those outside the scope of the carousel, are shifted left a little. This doesn't occur when I just have plain images in the carousel. Any idea what this could be? I'll try to make a minimal example

**Comment by ghost** [comment]
+1, came here looking for this. It would be nice to get the slot renamed and aliased for compatability.

Better docs would also be good for non-image content. I'll do a PR when I get some time.
